syntax = "proto3";

package foreverbull.backtest;

option go_package = "github.com/lhjnilsson/foreverbull/pkg/pb/backtest";

import "foreverbull/backtest/ingestion.proto";
import "google/protobuf/timestamp.proto";

message GetCurrentIngestionRequest {
}

message GetCurrentIngestionResponse {
    Ingestion ingestion = 1;
    IngestionStatus status = 2;
    int64 size = 3;
}

message UpdateIngestionRequest {
}

message UpdateIngestionResponse {
    Ingestion ingestion = 1;
    IngestionStatus status = 2;
    string errorMessage = 3;
}

service IngestionServicer {
    rpc GetCurrentIngestion(GetCurrentIngestionRequest) returns (GetCurrentIngestionResponse);
    rpc UpdateIngestion(UpdateIngestionRequest) returns (stream UpdateIngestionResponse);
}
